Optioanl tuple cuases unexpected C error

OS: linux, Ubuntu 20.04.3 LTS
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-1035G1 CPU @ 1.00GHz
CC version: cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

getwd: /home/chaos-unity/桌面/projects/voracity
vmodules: /home/chaos-unity/.vmodules
vroot: /home/chaos-unity/桌面/projects/v
vexe: /home/chaos-unity/桌面/projects/v/v
vexe mtime: 2022-04-06 03:36:06
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 f75ce9d.6412f8b

Git version: git version 2.25.1
Git vroot status: weekly.2022.13-69-g6412f8ba
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

**What did you do?**
```v
fn tuple() ?(int, int) {
	return 1, 1
}

fn main() {
	println(tuple()?)
}
```

**What did you expect to see?**
It should print `(1, 1)`
 
**What did you see instead?**
```cmd
==================
/tmp/v_1000/main.7627588759998129153.tmp.c:11450: error: cannot convert 'struct Option_multi_return_int_int' to 'struct multi_return_int_int'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

## Update 1
This error seems to be only occurred when the calling is suffixed with `?`

## Update 2
This error is caused by calling multi return type's `str` function

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -471,7 +471,7 @@ fn (mut g Gen) gen_multi_return_assign(node &ast.AssignStmt, return_type ast.Typ
 	// TODO Handle in if_expr
 	is_opt := return_type.has_flag(.optional)
 	mr_var_name := 'mr_$node.pos.pos'
-	mr_styp := g.typ(return_type)
+	mr_styp := g.typ(return_type.clear_flag(.optional))
 	g.write('$mr_styp $mr_var_name = ')
 	g.expr(node.right[0])
 	g.writeln(';')
@@ -505,9 +505,9 @@ fn (mut g Gen) gen_multi_return_assign(node &ast.AssignStmt, return_type ast.Typ
 			if is_opt {
 				mr_base_styp := g.base_type(return_type)
 				if is_auto_heap {
-					g.writeln('HEAP${noscan}($mr_base_styp, *($mr_base_styp*)${mr_var_name}.data).arg$i) });')
+					g.writeln('HEAP${noscan}($mr_base_styp, ${mr_var_name}.arg$i) });')
 				} else {
-					g.writeln('(*($mr_base_styp*)${mr_var_name}.data).arg$i });')
+					g.writeln('${mr_var_name}.arg$i });')
 				}
 			} else {
 				if is_auto_heap {
@@ -520,9 +520,9 @@ fn (mut g Gen) gen_multi_return_assign(node &ast.AssignStmt, return_type ast.Typ
 			if is_opt {
 				mr_base_styp := g.base_type(return_type)
 				if is_auto_heap {
-					g.writeln(' = HEAP${noscan}($mr_base_styp, *($mr_base_styp*)${mr_var_name}.data).arg$i);')
+					g.writeln(' = HEAP${noscan}($mr_base_styp, ${mr_var_name}.arg$i);')
 				} else {
-					g.writeln(' = (*($mr_base_styp*)${mr_var_name}.data).arg$i;')
+					g.writeln(' = ${mr_var_name}.arg$i;')
 				}
 			} else {
 				if is_auto_heap {

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -695,8 +695,6 @@ fn (mut g Gen) call_expr(node ast.CallExpr) {
 		unwrapped_styp := g.typ(unwrapped_typ)
 		if unwrapped_typ == ast.void_type {
 			g.write('\n $cur_line')
-		} else if g.table.sym(node.return_type).kind == .multi_return {
-			g.write('\n $cur_line $tmp_opt /*U*/')
 		} else {
 			if !g.inside_const_optional {
 				g.write('\n $cur_line (*($unwrapped_styp*)${tmp_opt}.data)')

--- a/vlib/v/tests/optional_multi_return_test.v
+++ b/vlib/v/tests/optional_multi_return_test.v
@@ -1,0 +1,10 @@
+fn tuple() ?(int, int) {
+	return 1, 2
+}
+
+fn test_optional_multi_return() ? {
+	println(tuple() ?)
+	a, b := tuple() ?
+	assert a == 1
+	assert b == 2
+}

